data/data/aws/vpc/master-elb.tf

```diff
@@ -41,8 +41,9 @@ resource "aws_lb_target_group" "api_internal" {
   ), var.tags)}"
 
   health_check {
-    healthy_threshold   = 3
-    unhealthy_threshold = 3
+    # 2+1 times interval must be smaller than 35 sec the kube-apiserver waits
```

Comment:
I'm having trouble with the `+1`.  From [here][1]:

> If the health checks exceed `UnhealthyThresholdCount` consecutive failures, the load balancer takes the target out of service.

So worst case scenario is a check right before the graceful shutdown begins.  30 seconds later we have our third unhealthy ping.  You're saying the target is currently waiting for the *next* unhealthy ping before [deregistering and draining][2]?  I think I need to do some experiments ;).

[1]: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/target-group-health-checks.html
[2]: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-target-groups.html#deregistration-delay

Reply:
@cuppett confirmed that this limit is "missing this many is ok, but one more kicks you out".

Reply:
I agree: +10, +20, +30 seconds => target taken out of the loop. This means we are left with 5 seconds to actually converge on that behaviour. Should be feasible.